### PR TITLE
Add support for volume Subpath option

### PR DESCRIPTION
**Pull request by vvoland**
- needs: https://github.com/docker/compose/pull/11583
- needs: https://github.com/compose-spec/compose-go/pull/594

**What I did**
Allows to use https://github.com/moby/moby/pull/45687 in compose

**Related issue**

**(not mandatory) A picture of a cute animal, if possible in relation to what you did**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -144,8 +141,9 @@ jobs:
           - plugin
           - standalone
         engine:
+          - 26.0.0-rc2
```
Comment (ndeloof) [inline .github/workflows/ci.yml:144]: 👍 we indeed should have tests ran with RC release before those get public

**Review by ndeloof — APPROVED** [review]

**Comment by ndeloof** [comment]
please rebase so we can merge

**Comment by vvoland** [comment]
We still need https://github.com/docker/compose/pull/11583 to be merged first

**Review by jhrotko — APPROVED** [review]
LGTM

**Comment by ndeloof** [comment]
I cherry-picked required changes into https://github.com/docker/compose/pull/11652

**Pull request closed without merging**